internal/services/compute/shared_image_gallery_data_source.go

```diff
@@ -90,5 +99,25 @@ func dataSourceSharedImageGalleryRead(d *pluginsdk.ResourceData, meta interface{
 		}
 	}
 
+	imagesResp, err := imagesClient.ListByGalleryComplete(ctx, id.ResourceGroupName, id.GalleryName)
+	if err != nil {
+		return fmt.Errorf("retrieving Shared Images ( Gallery %q / Resource Group %q): %+v", id.GalleryName, id.ResourceGroupName, err)
+	}
+
+	images := make([]compute.GalleryImage, 0)
+	for imagesResp.NotDone() {
+		images = append(images, imagesResp.Value())
+		if err := imagesResp.NextWithContext(ctx); err != nil {
+			return fmt.Errorf("listing next page of shared images for Gallery %q / Resource Group %q: %+v", id.GalleryName, id.ResourceGroupName, err)
+		}
+	}
+
+	imageNames := make([]string, 0)
+	for _, image := range images {
+		imageNames = append(imageNames, *image.Name)
+	}
```

Comment:
I think this can be simplified, we should also nil check the Name since it's a pointer

```suggestion
	imageNames := make([]string, 0)
	for imagesResp.NotDone() {
		image := imagesResp.Value()
		if image.Name != nil {
			imageNames = append(imageNames, *imagesResp.Value().Name)
		}	
		if err := imagesResp.NextWithContext(ctx); err != nil {
			return fmt.Errorf("listing next page of shared images for %s: %+v", id, err)
		}
	}
```